aptos-move/framework/src/natives/transaction_context.rs

```diff
@@ -111,6 +130,91 @@ fn native_get_script_hash(
     )])
 }
 
+fn native_sender_internal(
+    context: &mut SafeNativeContext,
+    mut _ty_args: Vec<Type>,
+    _args: VecDeque<Value>,
+) -> SafeNativeResult<SmallVec<[Value; 1]>> {
+    context.charge(TRANSACTION_CONTEXT_SENDER_BASE)?;
+
+    let transaction_context_opt = get_transaction_context_opt_from_context(context);
+    if let Some(transaction_context) = transaction_context_opt {
+        Ok(smallvec![Value::address(transaction_context.sender())])
+    } else {
+        Err(SafeNativeError::Abort {
+            abort_code: error::invalid_state(abort_codes::E_TRANSACTION_CONTEXT_NOT_AVAILABLE),
+        })
+    }
+}
+
+fn native_secondary_signers_internal(
+    context: &mut SafeNativeContext,
+    mut _ty_args: Vec<Type>,
+    _args: VecDeque<Value>,
+) -> SafeNativeResult<SmallVec<[Value; 1]>> {
+    context.charge(TRANSACTION_CONTEXT_SECONDARY_SIGNERS_BASE)?;
+
+    let transaction_context_opt = get_transaction_context_opt_from_context(context);
+    if let Some(transaction_context) = transaction_context_opt {
+        let secondary_signers = transaction_context.secondary_signers();
+        context.charge(
+            TRANSACTION_CONTEXT_SECONDARY_SIGNERS_PER_SIGNER
+                * NumArgs::new(secondary_signers.len() as u64),
+        )?;
+        Ok(smallvec![Value::vector_address(secondary_signers)])
+    } else {
+        Err(SafeNativeError::Abort {
+            abort_code: error::invalid_state(abort_codes::E_TRANSACTION_CONTEXT_NOT_AVAILABLE),
+        })
+    }
+}
+
+fn native_fee_payer_opt_internal(
+    context: &mut SafeNativeContext,
+    mut _ty_args: Vec<Type>,
+    _args: VecDeque<Value>,
+) -> SafeNativeResult<SmallVec<[Value; 1]>> {
+    context.charge(TRANSACTION_CONTEXT_FEE_PAYER_BASE)?;
+
+    let transaction_context_opt = get_transaction_context_opt_from_context(context);
+    if let Some(transaction_context) = transaction_context_opt {
+        let struct_value = Struct::pack(vec![Value::vector_address(
+            transaction_context.fee_payer_opt(),
+        )]);
+        Ok(smallvec![Value::struct_(struct_value)])
+    } else {
+        Err(SafeNativeError::Abort {
+            abort_code: error::invalid_state(abort_codes::E_TRANSACTION_CONTEXT_NOT_AVAILABLE),
+        })
+    }
+}
+
+fn native_max_gas_amount_internal(
```

Comment:
What units is this in?

Reply:
Yes, 'max_gas_amount'  is in Octa: https://aptos.dev/concepts/txns-states/#transactions

Reply:
Sure, I will put up a PR shortly.